front-end/src/search/posts.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect} from "react";
+import { Link, useNavigate, useParams } from "react-router-dom";
+import {baseURL, client} from "../config/AxiosConfig";
+import { decodeToken } from "react-jwt";
+import Cookies from 'universal-cookie';
+
+const Posts = () => {
+
+  const navigate = useNavigate();
+  const [posts, setPosts] = useState<any[]>([]);
+  const params = useParams();
+
+  const cookies = new Cookies();
+  let user_token = cookies.get('user_token')
+
+  const isTokenExpired= async () => {
+    const headers = {'Authorization': 'Bearer '+user_token}
+    const response: any = await client.get(baseURL+`/users/token_expired`,{headers: headers});
+    if(response.status==200 && response.data.response=="expired"){
+      cookies.set('user_token', response.data.new_token, { path: '/' });
+    }
+  }
+
+  const likePost = async (post:any) => {
+    await isTokenExpired();
+    const decodeUserToken = JSON.parse( JSON.stringify(decodeToken(user_token+"")));
+    const payload = {post_id:post.id, user_id: decodeUserToken.user_id, token: user_token};
+    const headers = {'Authorization': 'Bearer '+ user_token}
+    const response: any = await client.post(baseURL+`/posts/${post.id}/likers/create`,payload,{headers: headers});
+    if(response.status==200 && response.data.result!=='liked') {
+      navigate("/posts")
+      alert('Post liked');
+      window.location.reload();
```

Comment:
why do we need this window.location?? if we are spa. it is not recommended to do this.